### PR TITLE
Fix super() in non derived class

### PR DESCRIPTION
**Pull request by eryue0220**
This pr solved the #2000 problem.

### CODE REVIEW

File: test/test-262-whitelist.txt [hunk 1]
```diff
@@ -17518,4 +17514,12 @@ test/language/statements/class/async-gen-method-static/forbidden-ext/b2/cls-decl
 test/language/statements/class/async-gen-method-static/forbidden-ext/b2/cls-decl-async-gen-meth-static-forbidden-ext-indirect-access-prop-caller.js(default)
 test/language/statements/class/async-gen-method-static/forbidden-ext/b2/cls-decl-async-gen-meth-static-forbidden-ext-indirect-access-own-prop-caller-value.js(default)
 test/language/statements/class/elements/syntax/valid/grammar-special-prototype-async-gen-meth-valid.js(default)
-test/language/statements/class/elements/syntax/valid/grammar-special-prototype-async-gen-meth-valid.js(strict mode)
\ No newline at end of file
+test/language/statements/class/elements/syntax/valid/grammar-special-prototype-async-gen-meth-valid.js(strict mode)
```
Comment (ljqx) [inline test/test-262-whitelist.txt:17517]: these tests are broken.

In general, `super()` is not allowed in non derived class, but `super.xxx` is allowed.

**Review by ljqx — CHANGES_REQUESTED** [review]

File: test/test-262-whitelist.txt (hunk 1 above)
Comment (eryue0220) [reply test/test-262-whitelist.txt:17517]: OK, I will check that.

Comment (eryue0220) [reply test/test-262-whitelist.txt:17517]: I run the test in the local env, it can pass.

Comment (ljqx) [reply test/test-262-whitelist.txt:17517]: Then you can remove these newly added tests from this whitelist file.

Comment (eryue0220) [reply test/test-262-whitelist.txt]: done.

**Comment by jogibear9988** [comment]
@eryue0220 

I think this test is still broken:

test/language/statements/class/syntax/class-body-method-definition-super-property.js
see: https://github.com/tc39/test262/blob/main/test/language/statements/class/syntax/class-body-method-definition-super-property.js

**Pull request closed without merging**